File: week1/day4/algo-anagrams-i/js/characterMatch.js
exports.isCharacterMatch = function(string1, string2) {

    let string_dict = {}

    string1 = string1.toLowerCase().replaceAll(' ', '')
    string2 = string2.toLowerCase().replaceAll(' ', '')

    // console.log(string1)
    // console.log(string2)

    for (let i = 0; i < string1.length; i++) {
        if (string1[i] in string_dict) {
            string_dict[string1[i]] += 1
        } else {
            string_dict[string1[i]] = 1
        }
    }

    for (let i = 0; i < string2.length; i++) {
        if (string2[i] in string_dict) {
            string_dict[string2[i]] -= 1
        } else {
            return false
        }
    }

    for (let i = 0; i < string_dict.length; i++) {
        if (string_dict[i] > 0) {
            return false
        }
    }

    return true

    console.log(string_dict)

};

File: week1/day4/algo-anagrams-i/python/character_match.py
# Don't forget to run your tests!

def is_character_match(string1, string2):
	letter_count = {}

	string1 = string1.lower()
	string2 = string2.lower()

	string1 = string1.replace(' ', '')
	string2 = string2.replace(' ', '')

	for char in string1:
		if char in letter_count:
			letter_count[char] += 1
		else:
			letter_count[char] = 1

	for char in string2:
		if char in letter_count:
			letter_count[char] -= 1
		else:
			return False

	# print(letter_count)

	for key in letter_count:
		if letter_count[key] > 0:
			return False
	
	return True

	# print(letter_count)